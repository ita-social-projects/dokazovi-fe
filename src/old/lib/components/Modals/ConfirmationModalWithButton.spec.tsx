import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  ConfirmationModalWithButton,
  IConfirmationModalWithButtonProps,
} from './ConfirmationModalWithButton';

const PROPS_MOCK: IConfirmationModalWithButtonProps = {
  message: 'Test Message',
  buttonText: 'Open Modal',
  buttonIcon: <div>Open Icon</div>,
  onConfirmButtonClick: () => null,
};

const renderAndOpenModal = () => {
  render(
    <ConfirmationModalWithButton
      message={PROPS_MOCK.message}
      buttonText={PROPS_MOCK.buttonText}
      onConfirmButtonClick={PROPS_MOCK.onConfirmButtonClick}
    />,
  );

  const openBtn = screen.getByText('Open Modal');
  userEvent.click(openBtn);
};

describe('ConfirmationModalWithButton', () => {
  beforeEach(() => renderAndOpenModal());

  it('should render correctly and match the snapshot', () => {
    const { asFragment } = render(
      <ConfirmationModalWithButton
        message={PROPS_MOCK.message}
        buttonText={PROPS_MOCK.buttonText}
        buttonIcon={PROPS_MOCK.buttonIcon}
        onConfirmButtonClick={PROPS_MOCK.onConfirmButtonClick}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display correct message when opened', () => {
    expect(screen.getByText('Test Message')).toBeInTheDocument();
  });

  it('should has confirmation button when opened', () => {
    expect(screen.getByText('Так')).toBeInTheDocument();
  });

  it('should has cancellation button when opened', () => {
    expect(screen.getByText('Ні')).toBeInTheDocument();
  });
});

describe('ConditionsNav Events', () => {
  it('should open when icon button icon is clicked', () => {
    const { container } = render(
      <ConfirmationModalWithButton
        message={PROPS_MOCK.message}
        buttonText={PROPS_MOCK.buttonText}
        buttonIcon={PROPS_MOCK.buttonIcon}
        onConfirmButtonClick={PROPS_MOCK.onConfirmButtonClick}
      />,
    );
    const openIcon = screen.getByText('Open Modal');
    userEvent.click(openIcon);

    expect(container).toMatchSnapshot();
  });

  it('should execute given function when confirmation button is clicked', () => {
    const confirmHandleMock = jest.fn();
    render(
      <ConfirmationModalWithButton
        message={PROPS_MOCK.message}
        buttonText={PROPS_MOCK.buttonText}
        onConfirmButtonClick={confirmHandleMock}
      />,
    );
    const openBtn = screen.getByText('Open Modal');
    userEvent.click(openBtn);

    const confirmationBtn = screen.getByText('Так');
    userEvent.click(confirmationBtn);

    expect(confirmHandleMock).toHaveBeenCalledTimes(1);
  });

  it('should close when decline button is clicked', () => {
    const { asFragment } = render(
      <ConfirmationModalWithButton
        message={PROPS_MOCK.message}
        buttonText={PROPS_MOCK.buttonText}
        onConfirmButtonClick={PROPS_MOCK.onConfirmButtonClick}
      />,
    );
    const openBtn = screen.getByText('Open Modal');
    userEvent.click(openBtn);

    const confirmationBtn = screen.getByText('Ні');
    userEvent.click(confirmationBtn);

    expect(asFragment()).toMatchSnapshot();
  });
});
