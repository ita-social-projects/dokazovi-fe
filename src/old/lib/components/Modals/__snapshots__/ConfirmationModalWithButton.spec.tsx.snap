// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionsNav Events should close when decline button is clicked 1`] = `
<DocumentFragment>
  <div
    data-testid="confirmation-modal"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Open Modal
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`ConditionsNav Events should open when icon button icon is clicked 1`] = `
<div
  aria-hidden="true"
>
  <div
    data-testid="confirmation-modal"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      style="padding: 0px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <div>
          Open Icon
        </div>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Open Modal
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ConfirmationModalWithButton should render correctly and match the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="confirmation-modal"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      style="padding: 0px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <div>
          Open Icon
        </div>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Open Modal
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
